### ISSUE TITLE
do_connect problem

### ISSUE THREAD

**Issue by kangyite** [mustafakemalgilor/tdslite#17]
This is not a problem when I connect to the wifi that have the sql server, but when I switch to my mobile hotspot this function will still return success as it will return 0 (Success code) even if cr is 0 which crash the code. So i added `if(cr == 0) return 1;` to prevent the problem
```
template <typename T>
        TDSL_SYMBOL_VISIBLE int do_connect(T target, tdsl::uint16_t port) {
            // Disconnect if already connected
            do_disconnect();

            int cr      = 0;
            int retries = 5;

            // There may be residue data in network buffer
            // Reset it to ensure a clean start
            this->network_buffer.get_writer()->reset();

            tdsl::char_view destination_host{};

            // Given that @p target can be a progmem string,
            // we need to ensure that it is in SRAM before passing
            // it to `connect` function. For that, instead of allocating
            // a new buffer, we use the network buffer since it's sufficiently
            // large free space not yet in use for anything.

            if (target.size_bytes() <= this->network_buffer.get_writer()->remaining_bytes()) {

                auto w = this->network_buffer.get_writer();
                for (auto c : target) {
                    auto wr = w->write(c);
                    (void) wr;
                }
                destination_host = w->inuse_span().template rebind_cast<const char>();
            }
            else {
                return -99; // FIXME: proper error code
            }

            // Retry up to MAX_CONNECT_ATTEMPTS times.
            while (retries--) {
                TDSL_DEBUG_PRINTLN("... attempting to connect to %s:%d, %d retries remaining ...",
                                   destination_host.data(), port, retries);
                cr = client.connect(destination_host.data(), port);
                if (cr == 1) {
                    TDSL_DEBUG_PRINTLN("... connected, %d --> %d.%d.%d.%d:%d ...",
                                       client.localPort(), client.remoteIP() [0],
                                       client.remoteIP() [1], client.remoteIP() [2],
                                       client.remoteIP() [3], client.remotePort());
                    break;
                }

                TDSL_DEBUG_PRINTLN("... connection attempt failed (%d) ...", cr);
                delay(1000);
            }

            // Reset the network buffer so it's ready to
            // use by its real purpose
            this->network_buffer.get_writer()->reset();

            if (cr == 1) {
                return 0;
            }
            if(cr == 0) return 1; // Added this then good
            return cr;
        }
```

**Comment by mustafakemalgilor** [comment]
Hello @kangyite,

Thanks for reporting this, the do_connect function needs to handle the `connect`'s return values better. I'll try address this in the next update.

**Comment by mustafakemalgilor** [comment]
Fixed with 0.4.0 release.
